TestProcessCollector failing on ppc64le architecture

I'm not quite clear why, but when trying to run the tests on little-endian PowerPC 64 host, three tests fail the assertion about memory-resident bytes.

Here is a log portion extrapolated from an RPM build done in SUSE's Open Build Service:

```
============================= test session starts ==============================
[  254s] platform linux2 -- Python 2.7.17, pytest-3.10.1, py-1.5.2, pluggy-0.8.0 -- /usr/bin/python2
[  254s] cachedir: .pytest_cache
[  254s] rootdir: /home/abuild/rpmbuild/BUILD/client_python-0.7.1, inifile:
[  266s] collecting ... collected 232 items
[  266s] 
[  266s] tests/test_core.py::TestCounter::test_block_decorator PASSED             [  0%]
[  266s] tests/test_core.py::TestCounter::test_function_decorator PASSED          [  0%]
[  266s] tests/test_core.py::TestCounter::test_increment PASSED                   [  1%]
[  266s] tests/test_core.py::TestCounter::test_negative_increment_raises PASSED   [  1%]
[  269s] tests/test_core.py::TestGauge::test_function_decorator_multithread PASSED [  2%]
[  269s] tests/test_core.py::TestGauge::test_gauge PASSED                         [  2%]
[  269s] tests/test_core.py::TestGauge::test_gauge_function PASSED                [  3%]
[  269s] tests/test_core.py::TestGauge::test_inprogress_block_decorator PASSED    [  3%]
[  269s] tests/test_core.py::TestGauge::test_inprogress_function_decorator PASSED [  3%]
[  269s] tests/test_core.py::TestGauge::test_time_block_decorator PASSED          [  4%]
[  269s] tests/test_core.py::TestGauge::test_time_function_decorator PASSED       [  4%]
[  269s] tests/test_core.py::TestSummary::test_block_decorator PASSED             [  5%]
[  269s] tests/test_core.py::TestSummary::test_function_decorator PASSED          [  5%]
[  269s] tests/test_core.py::TestSummary::test_function_decorator_multithread PASSED [  6%]
[  269s] tests/test_core.py::TestSummary::test_function_decorator_reentrancy PASSED [  6%]
[  269s] tests/test_core.py::TestSummary::test_summary PASSED                     [  6%]
[  269s] tests/test_core.py::TestHistogram::test_block_decorator PASSED           [  7%]
[  269s] tests/test_core.py::TestHistogram::test_function_decorator PASSED        [  7%]
[  270s] tests/test_core.py::TestHistogram::test_function_decorator_multithread PASSED [  8%]
[  270s] tests/test_core.py::TestHistogram::test_histogram PASSED                 [  8%]
[  270s] tests/test_core.py::TestHistogram::test_labels PASSED                    [  9%]
[  270s] tests/test_core.py::TestHistogram::test_setting_buckets PASSED           [  9%]
[  270s] tests/test_core.py::TestInfo::test_info PASSED                           [  9%]
[  270s] tests/test_core.py::TestInfo::test_labels PASSED                         [ 10%]
[  270s] tests/test_core.py::TestEnum::test_enum PASSED                           [ 10%]
[  270s] tests/test_core.py::TestEnum::test_labels PASSED                         [ 11%]
[  270s] tests/test_core.py::TestEnum::test_overlapping_labels PASSED             [ 11%]
[  270s] tests/test_core.py::TestMetricWrapper::test_child PASSED                 [ 12%]
[  270s] tests/test_core.py::TestMetricWrapper::test_empty_labels_list PASSED     [ 12%]
[  270s] tests/test_core.py::TestMetricWrapper::test_incorrect_label_count_raises PASSED [ 12%]
[  270s] tests/test_core.py::TestMetricWrapper::test_invalid_names_raise PASSED   [ 13%]
[  270s] tests/test_core.py::TestMetricWrapper::test_labels_by_kwarg PASSED       [ 13%]
[  270s] tests/test_core.py::TestMetricWrapper::test_labels_coerced_to_string PASSED [ 14%]
[  270s] tests/test_core.py::TestMetricWrapper::test_labels_on_labels PASSED      [ 14%]
[  270s] tests/test_core.py::TestMetricWrapper::test_namespace_subsystem_concatenated PASSED [ 15%]
[  270s] tests/test_core.py::TestMetricWrapper::test_no_units_for_info_enum PASSED [ 15%]
[  270s] tests/test_core.py::TestMetricWrapper::test_non_string_labels_raises PASSED [ 15%]
[  270s] tests/test_core.py::TestMetricWrapper::test_remove PASSED                [ 16%]
[  270s] tests/test_core.py::TestMetricWrapper::test_unit_appended PASSED         [ 16%]
[  270s] tests/test_core.py::TestMetricWrapper::test_unit_notappended PASSED      [ 17%]
[  270s] tests/test_core.py::TestMetricFamilies::test_bad_constructors PASSED     [ 17%]
[  270s] tests/test_core.py::TestMetricFamilies::test_counter PASSED              [ 18%]
[  270s] tests/test_core.py::TestMetricFamilies::test_counter_labels PASSED       [ 18%]
[  270s] tests/test_core.py::TestMetricFamilies::test_counter_total PASSED        [ 18%]
[  270s] tests/test_core.py::TestMetricFamilies::test_gauge PASSED                [ 19%]
[  270s] tests/test_core.py::TestMetricFamilies::test_gauge_labels PASSED         [ 19%]
[  270s] tests/test_core.py::TestMetricFamilies::test_gaugehistogram PASSED       [ 20%]
[  270s] tests/test_core.py::TestMetricFamilies::test_gaugehistogram_labels PASSED [ 20%]
[  270s] tests/test_core.py::TestMetricFamilies::test_histogram PASSED            [ 21%]
[  270s] tests/test_core.py::TestMetricFamilies::test_histogram_labels PASSED     [ 21%]
[  270s] tests/test_core.py::TestMetricFamilies::test_info PASSED                 [ 21%]
[  270s] tests/test_core.py::TestMetricFamilies::test_info_labels PASSED          [ 22%]
[  270s] tests/test_core.py::TestMetricFamilies::test_labelnames PASSED           [ 22%]
[  270s] tests/test_core.py::TestMetricFamilies::test_stateset PASSED             [ 23%]
[  270s] tests/test_core.py::TestMetricFamilies::test_stateset_labels PASSED      [ 23%]
[  270s] tests/test_core.py::TestMetricFamilies::test_summary PASSED              [ 24%]
[  270s] tests/test_core.py::TestMetricFamilies::test_summary_labels PASSED       [ 24%]
[  270s] tests/test_core.py::TestMetricFamilies::test_untyped PASSED              [ 25%]
[  270s] tests/test_core.py::TestMetricFamilies::test_untyped_labels PASSED       [ 25%]
[  270s] tests/test_core.py::TestMetricFamilies::test_untyped_unit PASSED         [ 25%]
[  270s] tests/test_core.py::TestCollectorRegistry::test_autodescribe_disabled_by_default PASSED [ 26%]
[  270s] tests/test_core.py::TestCollectorRegistry::test_duplicate_metrics_raises PASSED [ 26%]
[  270s] tests/test_core.py::TestCollectorRegistry::test_restricted_registry PASSED [ 27%]
[  270s] tests/test_core.py::TestCollectorRegistry::test_unregister_works PASSED  [ 27%]
[  270s] tests/test_exposition.py::TestGenerateText::test_counter PASSED          [ 28%]
[  270s] tests/test_exposition.py::TestGenerateText::test_counter_total PASSED    [ 28%]
[  270s] tests/test_exposition.py::TestGenerateText::test_enum PASSED             [ 28%]
[  270s] tests/test_exposition.py::TestGenerateText::test_escaping PASSED         [ 29%]
[  270s] tests/test_exposition.py::TestGenerateText::test_gauge PASSED            [ 29%]
[  270s] tests/test_exposition.py::TestGenerateText::test_gaugehistogram PASSED   [ 30%]
[  270s] tests/test_exposition.py::TestGenerateText::test_histogram PASSED        [ 30%]
[  270s] tests/test_exposition.py::TestGenerateText::test_info PASSED             [ 31%]
[  270s] tests/test_exposition.py::TestGenerateText::test_nonnumber PASSED        [ 31%]
[  270s] tests/test_exposition.py::TestGenerateText::test_summary PASSED          [ 31%]
[  270s] tests/test_exposition.py::TestGenerateText::test_timestamp PASSED        [ 32%]
[  270s] tests/test_exposition.py::TestGenerateText::test_unicode PASSED          [ 32%]
[  270s] tests/test_exposition.py::TestPushGateway::test_delete PASSED            [ 33%]
[  270s] tests/test_exposition.py::TestPushGateway::test_delete_with_groupingkey PASSED [ 33%]
[  270s] tests/test_exposition.py::TestPushGateway::test_instance_ip_grouping_key PASSED [ 34%]
[  270s] tests/test_exposition.py::TestPushGateway::test_metrics_handler PASSED   [ 34%]
[  270s] tests/test_exposition.py::TestPushGateway::test_metrics_handler_subclassing PASSED [ 34%]
[  270s] tests/test_exposition.py::TestPushGateway::test_push PASSED              [ 35%]
[  270s] tests/test_exposition.py::TestPushGateway::test_push_with_basic_auth_handler PASSED [ 35%]
[  270s] tests/test_exposition.py::TestPushGateway::test_push_with_complex_groupingkey PASSED [ 36%]
[  270s] tests/test_exposition.py::TestPushGateway::test_push_with_groupingkey PASSED [ 36%]
[  270s] tests/test_exposition.py::TestPushGateway::test_push_with_handler PASSED [ 37%]
[  270s] tests/test_exposition.py::TestPushGateway::test_pushadd PASSED           [ 37%]
[  270s] tests/test_exposition.py::TestPushGateway::test_pushadd_with_groupingkey PASSED [ 37%]
[  270s] tests/test_exposition.py::test_basic_metric_families[None-TypeError-CounterMetricFamily] PASSED [ 38%]
[  270s] tests/test_exposition.py::test_basic_metric_families[None-TypeError-GaugeMetricFamily] PASSED [ 38%]
[  270s] tests/test_exposition.py::test_basic_metric_families[-ValueError-CounterMetricFamily] PASSED [ 39%]
[  270s] tests/test_exposition.py::test_basic_metric_families[-ValueError-GaugeMetricFamily] PASSED [ 39%]
[  270s] tests/test_exposition.py::test_basic_metric_families[x-ValueError-CounterMetricFamily] PASSED [ 40%]
[  270s] tests/test_exposition.py::test_basic_metric_families[x-ValueError-GaugeMetricFamily] PASSED [ 40%]
[  270s] tests/test_exposition.py::test_basic_metric_families[value3-TypeError-CounterMetricFamily] PASSED [ 40%]
[  270s] tests/test_exposition.py::test_basic_metric_families[value3-TypeError-GaugeMetricFamily] PASSED [ 41%]
[  270s] tests/test_exposition.py::test_basic_metric_families[value4-TypeError-CounterMetricFamily] PASSED [ 41%]
[  270s] tests/test_exposition.py::test_basic_metric_families[value4-TypeError-GaugeMetricFamily] PASSED [ 42%]
[  270s] tests/test_exposition.py::test_summary_metric_family[None-0-TypeError] PASSED [ 42%]
[  270s] tests/test_exposition.py::test_summary_metric_family[0-None-TypeError] PASSED [ 43%]
[  270s] tests/test_exposition.py::test_summary_metric_family[-0-ValueError] PASSED [ 43%]
[  270s] tests/test_exposition.py::test_summary_metric_family[0--ValueError] PASSED [ 43%]
[  270s] tests/test_exposition.py::test_summary_metric_family[count_value4-0-TypeError] PASSED [ 44%]
[  270s] tests/test_exposition.py::test_summary_metric_family[0-sum_value5-TypeError] PASSED [ 44%]
[  270s] tests/test_exposition.py::test_summary_metric_family[count_value6-0-TypeError] PASSED [ 45%]
[  270s] tests/test_exposition.py::test_summary_metric_family[0-sum_value7-TypeError] PASSED [ 45%]
[  270s] tests/test_exposition.py::test_histogram_metric_families[buckets0-None-TypeError-HistogramMetricFamily] PASSED [ 46%]
[  270s] tests/test_exposition.py::test_histogram_metric_families[buckets0-None-TypeError-GaugeHistogramMetricFamily] PASSED [ 46%]
[  270s] tests/test_exposition.py::test_histogram_metric_families[buckets1-0-TypeError-HistogramMetricFamily] PASSED [ 46%]
[  270s] tests/test_exposition.py::test_histogram_metric_families[buckets1-0-TypeError-GaugeHistogramMetricFamily] PASSED [ 47%]
[  270s] tests/test_exposition.py::test_histogram_metric_families[buckets2-0-AttributeError-HistogramMetricFamily] PASSED [ 47%]
[  270s] tests/test_exposition.py::test_histogram_metric_families[buckets2-0-AttributeError-GaugeHistogramMetricFamily] PASSED [ 48%]
[  270s] tests/test_exposition.py::test_histogram_metric_families[buckets3-0-TypeError-HistogramMetricFamily] PASSED [ 48%]
[  270s] tests/test_exposition.py::test_histogram_metric_families[buckets3-0-TypeError-GaugeHistogramMetricFamily] PASSED [ 49%]
[  270s] tests/test_exposition.py::test_histogram_metric_families[buckets4-0-AttributeError-HistogramMetricFamily] PASSED [ 49%]
[  270s] tests/test_exposition.py::test_histogram_metric_families[buckets4-0-AttributeError-GaugeHistogramMetricFamily] PASSED [ 50%]
[  270s] tests/test_exposition.py::test_histogram_metric_families[buckets5--ValueError-HistogramMetricFamily] PASSED [ 50%]
[  270s] tests/test_exposition.py::test_histogram_metric_families[buckets5--ValueError-GaugeHistogramMetricFamily] PASSED [ 50%]
[  270s] tests/test_exposition.py::test_histogram_metric_families[buckets6-0-ValueError-HistogramMetricFamily] PASSED [ 51%]
[  270s] tests/test_exposition.py::test_histogram_metric_families[buckets6-0-ValueError-GaugeHistogramMetricFamily] PASSED [ 51%]
[  270s] tests/test_exposition.py::test_histogram_metric_families[buckets7-0-ValueError-HistogramMetricFamily] PASSED [ 52%]
[  270s] tests/test_exposition.py::test_histogram_metric_families[buckets7-0-ValueError-GaugeHistogramMetricFamily] PASSED [ 52%]
[  270s] tests/test_gc_collector.py::TestGCCollector::test_empty SKIPPED          [ 53%]
[  270s] tests/test_gc_collector.py::TestGCCollector::test_working SKIPPED        [ 53%]
[  270s] tests/test_graphite_bridge.py::TestGraphiteBridge::test_labels PASSED    [ 53%]
[  270s] tests/test_graphite_bridge.py::TestGraphiteBridge::test_nolabels PASSED  [ 54%]
[  270s] tests/test_graphite_bridge.py::TestGraphiteBridge::test_prefix PASSED    [ 54%]
[  270s] tests/test_graphite_bridge.py::TestGraphiteBridge::test_sanitizing PASSED [ 55%]
[  270s] tests/test_multiprocess.py::TestMultiProcess::test_collect PASSED        [ 55%]
[  270s] tests/test_multiprocess.py::TestMultiProcess::test_counter_across_forks PASSED [ 56%]
[  270s] tests/test_multiprocess.py::TestMultiProcess::test_counter_adds PASSED   [ 56%]
[  270s] tests/test_multiprocess.py::TestMultiProcess::test_gauge_all PASSED      [ 56%]
[  270s] tests/test_multiprocess.py::TestMultiProcess::test_gauge_liveall PASSED  [ 57%]
[  270s] tests/test_multiprocess.py::TestMultiProcess::test_gauge_livesum PASSED  [ 57%]
[  270s] tests/test_multiprocess.py::TestMultiProcess::test_gauge_max PASSED      [ 58%]
[  270s] tests/test_multiprocess.py::TestMultiProcess::test_gauge_min PASSED      [ 58%]
[  270s] tests/test_multiprocess.py::TestMultiProcess::test_histogram_adds PASSED [ 59%]
[  270s] tests/test_multiprocess.py::TestMultiProcess::test_initialization_detects_pid_change PASSED [ 59%]
[  270s] tests/test_multiprocess.py::TestMultiProcess::test_merge_no_accumulate PASSED [ 59%]
[  270s] tests/test_multiprocess.py::TestMultiProcess::test_missing_gauge_file_during_merge PASSED [ 60%]
[  270s] tests/test_multiprocess.py::TestMultiProcess::test_namespace_subsystem PASSED [ 60%]
[  270s] tests/test_multiprocess.py::TestMultiProcess::test_summary_adds PASSED   [ 61%]
[  270s] tests/test_multiprocess.py::TestMmapedDict::test_corruption_detected PASSED [ 61%]
[  270s] tests/test_multiprocess.py::TestMmapedDict::test_expansion PASSED        [ 62%]
[  271s] tests/test_multiprocess.py::TestMmapedDict::test_multi_expansion PASSED  [ 62%]
[  271s] tests/test_multiprocess.py::TestMmapedDict::test_process_restart PASSED  [ 62%]
[  271s] tests/test_multiprocess.py::TestUnsetEnv::test_file_syncpath PASSED      [ 63%]
[  271s] tests/test_multiprocess.py::TestUnsetEnv::test_unset_syncdir_env PASSED  [ 63%]
[  271s] tests/test_parser.py::TestParse::test_blank_lines_and_comments PASSED    [ 64%]
[  271s] tests/test_parser.py::TestParse::test_commas PASSED                      [ 64%]
[  271s] tests/test_parser.py::TestParse::test_empty_brackets PASSED              [ 65%]
[  271s] tests/test_parser.py::TestParse::test_empty_help PASSED                  [ 65%]
[  271s] tests/test_parser.py::TestParse::test_empty_label PASSED                 [ 65%]
[  271s] tests/test_parser.py::TestParse::test_escaping PASSED                    [ 66%]
[  271s] tests/test_parser.py::TestParse::test_help_escaping PASSED               [ 66%]
[  271s] tests/test_parser.py::TestParse::test_label_escaping PASSED              [ 67%]
[  271s] tests/test_parser.py::TestParse::test_labels_and_infinite PASSED         [ 67%]
[  271s] tests/test_parser.py::TestParse::test_labels_with_curly_braces PASSED    [ 68%]
[  271s] tests/test_parser.py::TestParse::test_multiple_trailing_commas PASSED    [ 68%]
[  271s] tests/test_parser.py::TestParse::test_nan PASSED                         [ 68%]
[  271s] tests/test_parser.py::TestParse::test_no_metadata PASSED                 [ 69%]
[  271s] tests/test_parser.py::TestParse::test_roundtrip PASSED                   [ 69%]
[  271s] tests/test_parser.py::TestParse::test_simple_counter PASSED              [ 70%]
[  271s] tests/test_parser.py::TestParse::test_simple_gauge PASSED                [ 70%]
[  271s] tests/test_parser.py::TestParse::test_simple_histogram PASSED            [ 71%]
[  271s] tests/test_parser.py::TestParse::test_simple_summary PASSED              [ 71%]
[  271s] tests/test_parser.py::TestParse::test_spaces PASSED                      [ 71%]
[  271s] tests/test_parser.py::TestParse::test_summary_quantiles PASSED           [ 72%]
[  271s] tests/test_parser.py::TestParse::test_tabs PASSED                        [ 72%]
[  271s] tests/test_parser.py::TestParse::test_timestamps_discarded PASSED        [ 73%]
[  271s] tests/test_parser.py::TestParse::test_type_help_switched PASSED          [ 73%]
[  271s] tests/test_parser.py::TestParse::test_untyped PASSED                     [ 74%]
[  271s] tests/test_platform_collector.py::TestPlatformCollector::test_python_info PASSED [ 74%]
[  271s] tests/test_platform_collector.py::TestPlatformCollector::test_system_info_java PASSED [ 75%]
[  271s] tests/test_process_collector.py::TestProcessCollector::test_namespace FAILED [ 75%]
[  271s] tests/test_process_collector.py::TestProcessCollector::test_working FAILED [ 75%]
[  271s] tests/test_process_collector.py::TestProcessCollector::test_working_584 FAILED [ 76%]
[  271s] tests/test_process_collector.py::TestProcessCollector::test_working_fake_pid PASSED [ 76%]
[  271s] tests/test_twisted.py::MetricsResourceTest::test_reports_metrics SKIPPED [ 77%]
[  271s] tests/openmetrics/test_exposition.py::TestGenerateText::test_counter PASSED [ 77%]
[  271s] tests/openmetrics/test_exposition.py::TestGenerateText::test_counter_total PASSED [ 78%]
[  271s] tests/openmetrics/test_exposition.py::TestGenerateText::test_enum PASSED [ 78%]
[  271s] tests/openmetrics/test_exposition.py::TestGenerateText::test_escaping PASSED [ 78%]
[  271s] tests/openmetrics/test_exposition.py::TestGenerateText::test_gauge PASSED [ 79%]
[  271s] tests/openmetrics/test_exposition.py::TestGenerateText::test_gaugehistogram PASSED [ 79%]
[  271s] tests/openmetrics/test_exposition.py::TestGenerateText::test_histogram PASSED [ 80%]
[  271s] tests/openmetrics/test_exposition.py::TestGenerateText::test_histogram_exemplar PASSED [ 80%]
[  272s] tests/openmetrics/test_exposition.py::TestGenerateText::test_info PASSED [ 81%]
[  272s] tests/openmetrics/test_exposition.py::TestGenerateText::test_nonhistogram_bucket_exemplar PASSED [ 81%]
[  272s] tests/openmetrics/test_exposition.py::TestGenerateText::test_nonhistogram_exemplar PASSED [ 81%]
[  272s] tests/openmetrics/test_exposition.py::TestGenerateText::test_nonnumber PASSED [ 82%]
[  272s] tests/openmetrics/test_exposition.py::TestGenerateText::test_summary PASSED [ 82%]
[  272s] tests/openmetrics/test_exposition.py::TestGenerateText::test_timestamp PASSED [ 83%]
[  272s] tests/openmetrics/test_exposition.py::TestGenerateText::test_unicode PASSED [ 83%]
[  272s] tests/openmetrics/test_parser.py::TestParse::test_duplicate_timestamps PASSED [ 84%]
[  272s] tests/openmetrics/test_parser.py::TestParse::test_empty_brackets PASSED  [ 84%]
[  272s] tests/openmetrics/test_parser.py::TestParse::test_empty_help PASSED      [ 84%]
[  272s] tests/openmetrics/test_parser.py::TestParse::test_empty_label PASSED     [ 85%]
[  272s] tests/openmetrics/test_parser.py::TestParse::test_empty_metadata PASSED  [ 85%]
[  272s] tests/openmetrics/test_parser.py::TestParse::test_escaping PASSED        [ 86%]
[  272s] tests/openmetrics/test_parser.py::TestParse::test_exemplars_with_hash_in_label_values PASSED [ 86%]
[  272s] tests/openmetrics/test_parser.py::TestParse::test_fallback_to_state_machine_label_parsing SKIPPED [ 87%]
[  272s] tests/openmetrics/test_parser.py::TestParse::test_float_gauge PASSED     [ 87%]
[  272s] tests/openmetrics/test_parser.py::TestParse::test_gaugehistogram_exemplars PASSED [ 87%]
[  272s] tests/openmetrics/test_parser.py::TestParse::test_hash_in_label_value PASSED [ 88%]
[  272s] tests/openmetrics/test_parser.py::TestParse::test_help_escaping PASSED   [ 88%]
[  272s] tests/openmetrics/test_parser.py::TestParse::test_histogram_exemplars PASSED [ 89%]
[  272s] tests/openmetrics/test_parser.py::TestParse::test_info_timestamps PASSED [ 89%]
[  272s] tests/openmetrics/test_parser.py::TestParse::test_invalid_float_input PASSED [ 90%]
[  272s] tests/openmetrics/test_parser.py::TestParse::test_invalid_input PASSED   [ 90%]
[  272s] tests/openmetrics/test_parser.py::TestParse::test_label_escaping PASSED  [ 90%]
[  272s] tests/openmetrics/test_parser.py::TestParse::test_labels_and_infinite PASSED [ 91%]
[  272s] tests/openmetrics/test_parser.py::TestParse::test_labels_with_curly_braces PASSED [ 91%]
[  272s] tests/openmetrics/test_parser.py::TestParse::test_nan PASSED             [ 92%]
[  272s] tests/openmetrics/test_parser.py::TestParse::test_nan_gauge PASSED       [ 92%]
[  272s] tests/openmetrics/test_parser.py::TestParse::test_no_metadata PASSED     [ 93%]
[  272s] tests/openmetrics/test_parser.py::TestParse::test_no_newline_after_eof PASSED [ 93%]
[  272s] tests/openmetrics/test_parser.py::TestParse::test_null_byte PASSED       [ 93%]
[  272s] tests/openmetrics/test_parser.py::TestParse::test_roundtrip PASSED       [ 94%]
[  272s] tests/openmetrics/test_parser.py::TestParse::test_simple_counter PASSED  [ 94%]
[  272s] tests/openmetrics/test_parser.py::TestParse::test_simple_gauge PASSED    [ 95%]
[  272s] tests/openmetrics/test_parser.py::TestParse::test_simple_gaugehistogram PASSED [ 95%]
[  272s] tests/openmetrics/test_parser.py::TestParse::test_simple_histogram PASSED [ 96%]
[  272s] tests/openmetrics/test_parser.py::TestParse::test_simple_info PASSED     [ 96%]
[  272s] tests/openmetrics/test_parser.py::TestParse::test_simple_stateset PASSED [ 96%]
[  272s] tests/openmetrics/test_parser.py::TestParse::test_simple_summary PASSED  [ 97%]
[  272s] tests/openmetrics/test_parser.py::TestParse::test_summary_quantiles PASSED [ 97%]
[  272s] tests/openmetrics/test_parser.py::TestParse::test_timestamps PASSED      [ 98%]
[  272s] tests/openmetrics/test_parser.py::TestParse::test_type_help_switched PASSED [ 98%]
[  272s] tests/openmetrics/test_parser.py::TestParse::test_uint64_counter PASSED  [ 99%]
[  272s] tests/openmetrics/test_parser.py::TestParse::test_unit_gauge PASSED      [ 99%]
[  272s] tests/openmetrics/test_parser.py::TestParse::test_untyped PASSED         [100%]
[  272s] 
[  272s] =================================== FAILURES ===================================
[  272s] _____________________ TestProcessCollector.test_namespace ______________________
[  272s] 
[  272s] self = <tests.test_process_collector.TestProcessCollector testMethod=test_namespace>
[  272s] 
[  272s]     def test_namespace(self):
[  272s]         collector = ProcessCollector(proc=self.test_proc, pid=lambda: 26231, registry=self.registry, namespace='n')
[  272s]         collector._ticks = 100
[  272s]     
[  272s]         self.assertEqual(17.21, self.registry.get_sample_value('n_process_cpu_seconds_total'))
[  272s]         self.assertEqual(56274944.0, self.registry.get_sample_value('n_process_virtual_memory_bytes'))
[  272s] >       self.assertEqual(8114176, self.registry.get_sample_value('n_process_resident_memory_bytes'))
[  272s] E       AssertionError: 8114176 != 129826816.0
[  272s] 
[  272s] tests/test_process_collector.py:32: AssertionError
[  272s] ______________________ TestProcessCollector.test_working _______________________
[  272s] 
[  272s] self = <tests.test_process_collector.TestProcessCollector testMethod=test_working>
[  272s] 
[  272s]     def test_working(self):
[  272s]         collector = ProcessCollector(proc=self.test_proc, pid=lambda: 26231, registry=self.registry)
[  272s]         collector._ticks = 100
[  272s]     
[  272s]         self.assertEqual(17.21, self.registry.get_sample_value('process_cpu_seconds_total'))
[  272s]         self.assertEqual(56274944.0, self.registry.get_sample_value('process_virtual_memory_bytes'))
[  272s] >       self.assertEqual(8114176, self.registry.get_sample_value('process_resident_memory_bytes'))
[  272s] E       AssertionError: 8114176 != 129826816.0
[  272s] 
[  272s] tests/test_process_collector.py:20: AssertionError
[  272s] ____________________ TestProcessCollector.test_working_584 _____________________
[  272s] 
[  272s] self = <tests.test_process_collector.TestProcessCollector testMethod=test_working_584>
[  272s] 
[  272s]     def test_working_584(self):
[  272s]         collector = ProcessCollector(proc=self.test_proc, pid=lambda: "584\n", registry=self.registry)
[  272s]         collector._ticks = 100
[  272s]     
[  272s]         self.assertEqual(0.0, self.registry.get_sample_value('process_cpu_seconds_total'))
[  272s]         self.assertEqual(10395648.0, self.registry.get_sample_value('process_virtual_memory_bytes'))
[  272s] >       self.assertEqual(634880, self.registry.get_sample_value('process_resident_memory_bytes'))
[  272s] E       AssertionError: 634880 != 10158080.0
[  272s] 
[  272s] tests/test_process_collector.py:44: AssertionError
[  272s] =============== 3 failed, 225 passed, 4 skipped in 16.75 seconds ===============
```

I realise it might not be necessarily easy for you folks to get a PPC box, but I if you had any suggestion on how to attempt a fix, I might be able to get ahold of one.

--- a/prometheus_client/process_collector.py
+++ b/prometheus_client/process_collector.py
@@ -31,6 +31,8 @@ class ProcessCollector(object):
         except (ValueError, TypeError, AttributeError, OSError):
             pass
 
+        self._pagesize = _PAGESIZE
+
         # This is used to test if we can access /proc.
         self._btime = 0
         try:
@@ -60,7 +62,7 @@ class ProcessCollector(object):
             vmem = GaugeMetricFamily(self._prefix + 'virtual_memory_bytes',
                                      'Virtual memory size in bytes.', value=float(parts[20]))
             rss = GaugeMetricFamily(self._prefix + 'resident_memory_bytes', 'Resident memory size in bytes.',
-                                    value=float(parts[21]) * _PAGESIZE)
+                                    value=float(parts[21]) * self._pagesize)
             start_time_secs = float(parts[19]) / self._ticks
             start_time = GaugeMetricFamily(self._prefix + 'start_time_seconds',
                                            'Start time of the process since unix epoch in seconds.',

--- a/tests/test_process_collector.py
+++ b/tests/test_process_collector.py
@@ -14,6 +14,7 @@ class TestProcessCollector(unittest.TestCase):
     def test_working(self):
         collector = ProcessCollector(proc=self.test_proc, pid=lambda: 26231, registry=self.registry)
         collector._ticks = 100
+        collector._pagesize = 4096
 
         self.assertEqual(17.21, self.registry.get_sample_value('process_cpu_seconds_total'))
         self.assertEqual(56274944.0, self.registry.get_sample_value('process_virtual_memory_bytes'))
@@ -26,6 +27,7 @@ class TestProcessCollector(unittest.TestCase):
     def test_namespace(self):
         collector = ProcessCollector(proc=self.test_proc, pid=lambda: 26231, registry=self.registry, namespace='n')
         collector._ticks = 100
+        collector._pagesize = 4096
 
         self.assertEqual(17.21, self.registry.get_sample_value('n_process_cpu_seconds_total'))
         self.assertEqual(56274944.0, self.registry.get_sample_value('n_process_virtual_memory_bytes'))
@@ -38,6 +40,7 @@ class TestProcessCollector(unittest.TestCase):
     def test_working_584(self):
         collector = ProcessCollector(proc=self.test_proc, pid=lambda: "584\n", registry=self.registry)
         collector._ticks = 100
+        collector._pagesize = 4096
 
         self.assertEqual(0.0, self.registry.get_sample_value('process_cpu_seconds_total'))
         self.assertEqual(10395648.0, self.registry.get_sample_value('process_virtual_memory_bytes'))
@@ -49,6 +52,7 @@ class TestProcessCollector(unittest.TestCase):
     def test_working_fake_pid(self):
         collector = ProcessCollector(proc=self.test_proc, pid=lambda: 123, registry=self.registry)
         collector._ticks = 100
+        collector._pagesize = 4096
 
         self.assertEqual(None, self.registry.get_sample_value('process_cpu_seconds_total'))
         self.assertEqual(None, self.registry.get_sample_value('process_virtual_memory_bytes'))

